test/upgrade/zaptest_buffer_test.go

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
Normally in Go, if a file name is ended with `_test`, that means this file contains Go test cases like `TestXXX(t *testing.T)`, any specific reason that we are not following that convention?

Reply:
I know that. I separated the test sources to multiple files, just because it's a lot of test code is written. Smaller files are better that one single file with 1000 lines of code in my mind.

Reply:
Got it, just noticed all the `xx_test.go` files are under `upgrade_test` package, so that sounds good!